frontend/hub/api/request.ts

```diff
@@ -179,3 +180,26 @@ export async function getHubRequest<ResponseBody extends object | TaskResponse>(
   };
   return hubRequestCommon<ResponseBody>(url, options);
 }
+
+/*
+Use this function carefuly, can be only used in situations where there are not tons of data
+*/
+export async function getHubAllItems<ResponseBody, ResponseItem>(
+  url: string,
+  settings: {
+    getData: (res: ResponseBody) => ResponseItem[] | undefined;
+    getNextUrl: (res: ResponseBody) => string | undefined;
+  }
+) {
+  let actualUrl = url;
+  let results: ResponseItem[] = [];
+
+  while (actualUrl) {
+    const data = await requestGet<ResponseBody>(actualUrl);
```

Comment:
We could use `hubRequestCommon` it is more suitable for HUB API peculiarities.

Could we limit the amount of data fetched to avoid an infinite loop?

Could we pass a `signal?: AbortSignal` to abort the cancel the requests in case user navigates away from the page, and so on?

Also, I looked at the https://github.com/ansible/ansible-ui/blob/10633ccb1fcb33179539bef92b2baa5f87b2f7d9/frontend/awx/common/useAwxGetAllPages.tsx#L2 Perhaps we could leverage useSWRInfinite to deal with pagination here as well.

Reply:
Yeah, I was thinking about some watchdog. However, we are waiting for the API to complete this as single Task. But we dont know when (and if) will be completed...